shotover/src/codec/cassandra.rs

```diff
@@ -343,6 +393,53 @@ impl CassandraDecoder {
             }
         }
     }
+
+    fn extract_envelopes_from_payload(&self, mut payload: BytesMut) -> Result<Vec<Message>> {
+        let mut envelopes: Vec<Message> = vec![];
+
+        while !payload.is_empty() {
+            let body_len = i32::from_be_bytes(payload[5..9].try_into().unwrap()) as usize;
+
+            let envelope_len = ENVELOPE_HEADER_LEN + body_len;
+
+            if envelope_len > payload.len() {
+                return Err(anyhow!(format!(
+                    "envelope length {} is longer than payload length {}",
+                    envelope_len,
+                    payload.len()
+                ),));
+            }
+
+            let envelope = payload.split_to(envelope_len);
+
+            tracing::debug!(
+                "{}: incoming cassandra message:\n{}",
+                self.direction,
+                pretty_hex::pretty_hex(&envelope)
+            );
+
+            envelopes.push(Message::from_bytes(
+                envelope.freeze(),
+                crate::message::ProtocolType::Cassandra {
+                    compression: Compression::None,
+                },
+            ));
+        }
+
+        Ok(envelopes)
+    }
+}
+
+fn header_crc_mismatch_error(computed_crc: i32, header_crc24: i32) -> anyhow::Error {
+    anyhow!(format!(
+        "Header CRC mismatch - expected {header_crc24}, found {computed_crc}."
+    ))
+}
+
+fn payload_crc_mismatch_error(computed_crc: u32, payload_crc32: u32) -> anyhow::Error {
+    anyhow!(format!(
+        "Payload CRC mismatch - read {payload_crc32}, computed {computed_crc}."
```

Comment:
lets make the expected/found and read/computed terminology consistent between these errors